grc/gui/Dialogs.py

```diff
@@ -452,3 +454,288 @@ def choose_editor(parent, config):
     if editor == '':
         Messages.send('>>> No editor selected.\n')
     return editor
+
+class TrustAnswer(IntEnum):
+    DO_NOT_TRUST = 1
+    TRUST_ONCE = 2
+    TRUST_FOREVER = 3
+
+def trust_prompt(parent, config, flowgraph, message):
+    """
+    Get user response from a trust prompt dialog.
+    Return false if user trusts the flow graph, else return true.
+    """
+    response = TrustPrompt(parent, flowgraph, message).run_and_destroy()
+
+    if response == TrustAnswer.TRUST_FOREVER:
+        files = config.get_trusted_files()
+        files.append(flowgraph.grc_file_path)
+        config.set_trusted_files(files)
+
+    return not response in (TrustAnswer.TRUST_ONCE, TrustAnswer.TRUST_FOREVER)
+
+class TrustPrompt(Gtk.Dialog):
+    """ Prompt the user to trust the flow graph (once or forever)."""
+
+    def __init__(self, parent, flowgraph, message):
+        """
+        TrustPrompt constructor.
+        Args:
+            flowgraph: the flowgraph which should be trusted
+            message: the message text displayed by the prompt
+        """
+        Gtk.Dialog.__init__(
+            self,
+            title='Do you trust this flowgraph?',
+            transient_for=parent,
+            modal=True,
+            destroy_with_parent=True,
+        )
+
+        buttons = (
+            'Trust once', TrustAnswer.TRUST_ONCE,
+            'Trust forever', TrustAnswer.TRUST_FOREVER,
+            Gtk.STOCK_CANCEL, TrustAnswer.DO_NOT_TRUST
+        )
+
+        self.flowgraph = flowgraph
+
+        self.add_buttons(*buttons)
+        self.dialog_height = 300
+        self.set_size_request(Constants.MIN_DIALOG_WIDTH, self.dialog_height)
+        self.set_border_width(10)
+        self.counter = 0
+
+        # show first so Gtk sets dialog width and wraps text accordingly
+        self.show_all()
+
+        message_label = Gtk.Label(message)
```

Comment:
I'm getting this warning:
```
/usr/local/lib/python3.9/site-packages/gnuradio/grc/gui/Dialogs.py:512: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/PyGObject/InitializerDeprecations
  message_label = Gtk.Label(message)
```

Reply:
Hmm, did not see that, thanks for pointing it out! I've made it a keyword arg so it should be fixed now.